src/components/WeatherCard.js

```diff
@@ -3,28 +3,36 @@ import React, { Component, PropTypes } from 'react';
 import CardSummary from './CardSummary';
 import CardFooter from './CardFooter';
 
-export default class WeatherCard extends Component {
+// This is awesome that you're seeing how components can be sliced.
+// I'm going to take this one step further and say what we want to
+// do here is make this a "functional" component. Meaning, it's a pure
+// function that accepts parameters. A functional component differs from
+// a normal stateful component in 1 major way, it does not contain state.
+// React has some big performance boosts planned around functional components
+// so it's wise to use them when you just have pure render components (like this one)
 
-    render() {
-        return (
-            <div className="tile" ref="tile">
 
-                {this.props.data.loaded ?
-                    <div className="tile-inner">
-                        <header className="city">
-                            <h1>{this.props.data.city}</h1>
-                        </header>
-                        <CardSummary data={this.props.data} />
-                        <CardFooter data={this.props.data} reloadView={this.props.reloadView} />
-                    </div>
-                    : <div className="tile-inner-loading"><p>Fetching weather data</p></div> }
+const WeatherCard = ({ data, reloadView }) => (
```

Comment:
Yep! Anytime it's just a render component, just make it functional. This means react no longer needs to run any diffs on state and it also doesn't need to process the life-cycle methods. It just checks if the leaf is different and then renders it.